### PR TITLE
Aws sdk integration

### PR DESCRIPTION
**Pull request by landonreed**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Add basic http endpoint for sending AWS API Gateway request history to OTP Admin Dashboard (see https://github.com/ibi-group/otp-admin-ui/pull/4).

### CODE REVIEW

File: src/main/java/org/opentripplanner/middleware/controllers/api/LogController.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import static org.opentripplanner.middleware.spark.Main.hasConfigProperty;
+import static org.opentripplanner.middleware.spark.Main.inTestEnvironment;
+
+/**
+ * Sets up HTTP endpoints for getting logging and request summary information from AWS Cloudwatch and API Gateway.
+ */
+public class LogController {
+    private static final Logger LOG = LoggerFactory.getLogger(LogController.class);
+    private static AmazonApiGateway gateway;
+    public static void register (Service spark, String apiPrefix) {
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/controllers/api/LogController.java:40]: Method comment

File: src/main/java/org/opentripplanner/middleware/controllers/api/LogController.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        if (!inTestEnvironment) {
+            // FIXME Only build API Gateway client if not testing to avoid Travis issues with AWS credentials.
+            gateway = gatewayBuilder.build();
+        } else {
+            LOG.warn("Not building API Gateway client.");
+        }
+        spark.get(apiPrefix + "/secure/logs", LogController::getUsageLogs, JsonUtils::toJson);
+    }
+
+    public static List<GetUsageResult> getUsageLogs(Request req, Response res) {
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/controllers/api/LogController.java:58]: Method comment

**Review by br648 — DISMISSED** [review]
Just a couple of methods that need comments. The rest looks good, in particular registering the spark end point within the controller. Perhaps we should be doing that with all the controllers?

File: src/main/java/org/opentripplanner/middleware/controllers/api/LogController.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        if (hasConfigProperty("AWS_PROFILE")) {
+            gatewayBuilder.withCredentials(new ProfileCredentialsProvider(getConfigPropertyAsText("AWS_PROFILE")));
+        }
+        if (!inTestEnvironment) {
+            // FIXME Only build API Gateway client if not testing to avoid Travis issues with AWS credentials.
+            gateway = gatewayBuilder.build();
+        } else {
+            LOG.warn("Not building API Gateway client.");
+        }
+        spark.get(apiPrefix + "/secure/logs", LogController::getUsageLogs, JsonUtils::toJson);
```
Comment (binh-dam-ibigroup) [inline src/main/java/org/opentripplanner/middleware/controllers/api/LogController.java:51]: A reminder for us to discuss the API documentation (it is currently embedded in the endpoint definition for `ApiController` subclasses).

File: src/main/java/org/opentripplanner/middleware/controllers/api/LogController.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    public static List<GetUsageResult> getUsageLogs(Request req, Response res) {
+        // TODO: how can we extract this usage plan ID? Perhaps this will just be stored along with the API key
+        //  information in the database table containing info on third party apps.
+        GetApiKeysRequest getApiKeysRequest = new GetApiKeysRequest();
+        GetApiKeysResult apiKeys = gateway.getApiKeys(getApiKeysRequest);
+        apiKeys.getItems();
```
Comment (binh-dam-ibigroup) [inline src/main/java/org/opentripplanner/middleware/controllers/api/LogController.java]: Do we intend to use any of the `apiKeys`? If not, remove these lines.

Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/controllers/api/LogController.java]: The broader question for this and your TODO comment @landonreed is how is the API key passed to AWS?

**Review by binh-dam-ibigroup — COMMENTED** [review]
I need to understand how the log requests work without us passing the API gateway key to AWS.

File: pom.xml [hunk 5]
```diff
@@ -251,5 +251,17 @@
             <artifactId>auth0</artifactId>
             <version>1.15.0</version>
         </dependency>
+
+        <!-- https://mvnrepository.com/artifact/com.amazonaws/aws-java-sdk -->
```
Comment (evansiroky) [inline pom.xml]: Can you add some more comments about what these are used for?

**Review by evansiroky — CHANGES_REQUESTED** [review]
See comment and thumbs up on others' comments.

File: src/main/java/org/opentripplanner/middleware/controllers/api/LogController.java (hunk 4 above)
Comment (landonreed) [reply src/main/java/org/opentripplanner/middleware/controllers/api/LogController.java]: Just added some changes here that removes these unused lines for now and allows filtering on a specific key. As we start work on the third party developer account, we can begin to make this connection a little more clear.

**Review by evansiroky — APPROVED** [review]

**Review by binh-dam-ibigroup — APPROVED** [review]

**Pull request merged**